### DISCUSSION TITLE
关于must_direct的疑问,放在dns->request中是否更合适?

### DISCUSSION THREAD

**Discussion by HurryYU** [daeuniverse/dae#529] [General]
我的理解是,配置文件主要分为`group`(负责定义出站),`dns`(负责dns分流处理),`route`(负责流量出站路由)
假设浏览器访问`www.google.com`, `dae`的策略配置为`domain`时,我的理解是:
1. 首先dns流量达到`dae`,`dae`根据`dns`中`request`配置的规则选择`上游dns`执行解析
2. `dae`将得到的ip地址返回给客户机
3. 客户机对此地址发起连接请求
4. `dae`能得到domain与ip,并根据`route`中的配置选择出站方式

`must_direct`文档的解释是`让DNS流量不经过dae`,但是是配置在`route`中的.
那么是否意味着在DNS阶段(即上述步骤1)之前,就需要先读取route里面的规则,来判断流量是否需要经过`dae`的`dns`处理
如果是这样,那么将`must_direct`放在`dns`中的`request`内配置是否也是可行并更合适的?

**Comment by mzz2017** [comment]
一个典型场景是 must 一般和 pname 联用，这样使得例如 dnsmasq 的流量全部放行不走 dae。

更好的一个理解是，pname(dnsmasq)->must_rules，这个must_rules是一个关键字，代表不对dnsmasq启用dns模块防止死循环

**Comment by mzz2017** [comment]
我觉得应该把->must_direct改为->direct(with_dns_traffic_accept)
